cmd/webhook/main.go

```diff
@@ -97,8 +97,8 @@ func main() {
 			v1alpha1.SchemeGroupVersion.WithKind("TaskRun"):          &v1alpha1.TaskRun{},
 			v1alpha1.SchemeGroupVersion.WithKind("PipelineRun"):      &v1alpha1.PipelineRun{},
 		},
-		Logger: logger,
-
+		Logger:                logger,
+		DisallowUnknownFields: true,
```

Comment:
For my own curiosity: is this a feature that got added somewhere recently, or something we could have been using all along?

Reply:
and p.s. if it's not new, do you happen to know why ppl looking at this issue before thought we couldn't easily fix it? (https://github.com/tektoncd/pipeline/issues/297#issuecomment-444568074)

Reply:
It was added with https://github.com/knative/pkg/pull/338, so ~4month ago :wink: